Databases apps - OpenTelemetry (S3M) (#1771)
SumoLogic/sumologic-documentation: docs/integrations/databases/opentelemetry/redis.md

## Log Types

This app supports logs for Redis in Cluster mode or Standalone mode. The Redis logs are generated in files as configured in the cluster node configuration files in `/etc/redis/redis.conf`. For more details on Redis logs, [click here](https://redislabs.com/ebook/part-2-core-concepts/chapter-5-using-redis-for-application-support/5-1-logging-to-redis/).

## Creating Fields in Sumo Logic for Redis

The following are [Fields](https://help.sumologic.com/docs/manage/fields/) that will be created as part of the Redis App install if not already present.

* **`db.cluster.name`** - User configured. Enter a name to identify this Redis cluster. This cluster name will be shown in the Sumo Logic dashboards.
* **`db.system`** - Has fixed value of redis.
* **`deployment.environment`** - User configured. This is the deployment environment where the Redis cluster resides. For example: dev, prod or qa.
* **`sumo.datasource`** - has fixed value of redis.

## Prerequisite

This section provides instructions for configuring log collection for Redis running on a non-Kubernetes environment. By default, Redis logs are stored in a log file.

Follow the instructions to set up log collection:

> 1. To configure the Redis log file, locate your local `[redis.conf](https://download.redis.io/redis-stable/redis.conf)` configuration file in the database directory. By default, Redis logs are stored in `/var/log/redis/redis-server.log`.

Reviewer: redis.conf is not coming up as an embedded link on redering.

> 1. To configure the Redis log file, locate your local `[redis.conf](https://download.redis.io/redis-stable/redis.conf)` configuration file in the database directory. By default, Redis logs are stored in `/var/log/redis/redis-server.log`.

Reviewer: redis.conf is not coming up as an embedded link on redering.

Author: @nmohantyctr ☝️

Author: Took care of this and the all comments for you, @nmohantyctr .

cc @stacykor @JainM6

1. After determining the location of conf file, open your `redis.conf` configuration file in a text editor to modify its logging parameters as such:
   1. Specify the server verbosity level. The value `loglevel` in conf file can be set to one of the following:
      * debug (a lot of information, useful for development/testing)
      * verbose (many rarely useful info, but not a mess like the debug level)
      * notice (moderately verbose, what you want in production probably)

>       * warning (only very important/critical messages are logged) loglevel notice

Author: @nmohantyctr - [comment from ](https://github.com/SumoLogic/sumologic-documentation/pull/1765#discussion_r1121481147)@JainM6 

I think it's better if we remove this parenthesis, and define the terms, a one-line description is enough.

Author: Agreed, @JainM6. I'll go ahead and implement this. I've also taken care of all of yours, Stacy's, and Alekh's feedback. Let me know if I've missed anything.

>       * warning (only very important/critical messages are logged) loglevel notice

Author: @angadrandhawa1 @sumoanema @arunpatyal  @stacykor , since we're writing for a developer audience, I don't think we have to define these. Let me know what you think.

    1. Specify the log file name. Also the empty string can be used to force Redis to log on the standard output. If you use the standard output for logging but daemonize, logs will be sent to `/dev/null` logfile.
1. Save the `redis.conf` file and restart the Redis server using the command: `sudo  service redis-server restart`.

> Once the logs are configured to be written to a local file, follow the steps below to configure the collection in Sumo Logic.

Reviewer 2: ```suggestion
Once the logs are configured to write to a local file, follow the steps below to configure the collection in Sumo Logic.
```

## Configuring Collection and Installing the Redis App

As part of setting up the collection process and app installation, you can select the App from **App Catalog** and click **Install App**. Follow the steps below:

### Step 1: Set up Collector

:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the option of using an existing Collector.
:::

If you want to create a new Collector, select the **Add a new Collector** option.

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command that can be executed in the machine that needs monitoring. Once executed, it will install the Sumo Logic OpenTelemetry Collector agent. <br/> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/collector-screenshot.png')} alt="Collector" />

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/collector-screenshot.png')} alt="Collector" />

Reviewer: Screenshot url is mentioned twice.

Author: @nmohantyctr ☝️

### Step 2: Configure integration

OpenTelemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to the Sumo Logic platform.

In this step, we will be configuring the yaml required for Redis Collection.

The log file path configured to capture redis logs must be given here.

The files are typically located in `/var/log/redis/redis-server.log`. If you are using a customized path, check the `[redis.conf](https://download.redis.io/redis-stable/redis.conf)` file for this information.

Click on the **Download YAML File** button to get the yaml file.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/configuration-screenshot.png')} alt="Configuration" />

### Step 3: Sending logs to Sumo Logic

> Once you have the yaml file downloaded in step 2, please follow the below steps based on your environment

Reviewer 2: ```suggestion
Once you have the yaml file downloaded in step 2, follow the below steps based on your environment
```

#### Linux

1. Copy the yaml at **`/etc/otelcol-sumo/conf.d/`** folder in the Redis instance which needs to be monitored.
2. Restart the otelcol-sumo process using the below command `sudo systemctl restart otelcol-sumo`

#### Mac

1. Copy the yaml at `/etc/otelcol-sumo/conf.d/` folder in the Redis instance which needs to be monitored.

> 2. Restart the otelcol-sumo process using the command `otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --conf "glob:/etc/otelcol-sumo/conf.d/*.yaml"`

Author: @sumoanema @angadrandhawa1 , shall I create a table for these options?

Author: Created a table.

> Sumo Logic will start receiving data from your host machine when the above command is successfully executed.

Reviewer 2: ```suggestion
Sumo Logic will receive data from your host machine when the above command is successfully executed.
```

> This will install the app and dashboards to your Sumo Logic org.

Author: Is this what you meant in your [comment](https://github.com/SumoLogic/sumologic-documentation/pull/1765#discussion_r1121461962)?

```suggestion
Click **Next** to install the app and dashboards to your Sumo Logic org.
```

Reviewer: Yes

Author: Thanks for confirming!

Results will be available after a while. It is important to note that each panel slowly and automatically fills with data matching the time range query received since the panel was created. Your data should appear in the dashboard within 20 minutes, and you will see complete graphs and maps.

## Sample Log Message

> Here's a sample log message in Non-Kubernetes environment.

Reviewer 2: ```suggestion
Here's a sample log message in a non-Kubernetes environment.
```

`5275:S 17 Mar 2021  19:13:38.138 * MASTER <-> REPLICA sync: Finished with success`

## Sample Query

This sample Query is from the Redis - Logs dashboard > Logs panel.

```sql

> Query String

Reviewer: Query string should be outside code block ?

Author: @nmohantyctr ☝️
